InnoViber.BLL/Services/MessageService.cs

```diff
@@ -3,11 +3,39 @@
 using InnoViber.BLL.Models;
 using InnoViber.DAL.Interfaces;
 using InnoViber.DAL.Models;
+using InnoViber.Domain.Providers;
+using InnoViber.Domain.Enums;
+using InnoViber.DAL.Data;
 
 namespace InnoViber.BLL.Services;
 
 public class MessageService : GenericService<MessageModel, MessageEntity>, IMessageService
 {
-    public MessageService(IMapper mapper, IMessageRepository repository) : base(mapper, repository)
-    { }
+    private readonly IDateTimeProvider _dateTimeProvider;
+    private readonly IMapper _mapper;
+    private readonly IGenericRepository<MessageEntity> _repository;
+
+    public MessageService(IMapper mapper, IMessageRepository repository, IDateTimeProvider dateTimeProvider) 
+        : base(mapper, repository)
+    { 
+        _dateTimeProvider = dateTimeProvider;
+        _mapper = mapper;
+        _repository = repository;
+    }
+
+    public new Task Create(MessageModel model, CancellationToken ct)
```

Comment:
use override instead of "new": https://stackoverflow.com/questions/1399127/difference-between-new-and-override